repos/fdbt-site/src/pages/editCarnetProperties.tsx

```diff
@@ -0,0 +1,120 @@
+                            </label>
+                            <span className="govuk-hint" id="edit-carnet-quantity-hint">
+                                Must be 2 or more
+                            </span>
+                        </>
+                        <input
+                            className="govuk-input govuk-input--width-6"
+                            name={'carnetQuantityInput'}
+                            data-non-numeric
+                            type="text"
+                            id={'edit-carnet-quantity'}
+                            aria-describedby="edit-carnet-quantity-hint"
+                            defaultValue={quantity || ''}
+                        />
+                    </div>
+                    <div className="govuk-!-margin-left-2 govuk-!-margin-right-2">
+                        <>
+                            <label className="govuk-label" htmlFor="edit-carnet-expiry">
+                                Carnet expiry
+                            </label>
+                            <span className="govuk-hint" id="edit-carnet-expiry-hint">
+                                e.g. 2 months
+                            </span>
+                        </>
+                        <ExpirySelector
+                            defaultDuration={expiryTime || undefined}
+                            defaultUnit={expiryUnit || undefined}
+                            quantityName={'carnetExpiryDurationInput'}
+                            quantityId={'edit-carnet-expiry-duration'}
+                            hintId="edit-carnet-expiry-hint"
+                            unitName={'carnetExpiryUnitInput'}
+                            unitId={'edit-carnet-expiry-unit'}
+                            carnet
+                            errors={errors}
+                            hideFormGroupError
+                        />
+                    </div>
+                </div>
+                <br />
+                <input type="submit" value="Continue" id="continue-button" className="govuk-button" />
+            </CsrfForm>
+        </FullColumnLayout>
+    );
+};
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: EditCarnetPropertiesProps } => {
+    const csrfToken = getCsrfToken(ctx);
+    const ticket = getSessionAttribute(ctx.req, MATCHING_JSON_ATTRIBUTE);
+    const errors: ErrorInfo[] = getSessionAttribute(ctx.req, EDIT_CARNET_PROPERTIES_ERROR) || [];
+
+    let carnetDetails: CarnetDetails | undefined;
+    if (ticket) {
+        const product = ticket.products[0];
+
+        if ('carnetDetails' in product) {
+            carnetDetails = product.carnetDetails;
+        } else {
+            throw new Error('carnetDetails is undefined');
+        }
+    } else {
+        throw new Error('Ticket is undefined');
+    }
+
+    if (!carnetDetails?.expiryUnit || !carnetDetails?.quantity) {
+        throw new Error('A property in carnetDetails is undefined');
```

Comment:
CarnetDetails (the type) doesnt look like it can have any undefined properties - if there is a potential for a property to be undefined, can we use a different type to describe it?